Add improve-page-1 to master

Aichnerc opened the pull request:
**Checklist:**

 <!--- Make sure your PR is documented and tested before submission. Put an `x` in all the boxes that apply: -->
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [ ] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?


**Type of Changes:**

 <!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)


**What is the current behavior? (link to any open issues here)**
- Currently, there is no functionality on the buttons.
- Currently, the mobile view is trash.
- Currently, the SEO integration does not include React-Helmet, sitemap and robots.txt.
- Currently, you can not send a message to us.
- Currently, you do not get a response when sending a message to us.


**What is the new behavior (if this is a feature change)?**
- SmoothScrolling and functionality has been added to every button.
- The mobile view has been improved.
- The SEO integration has been improved.
- The contact form now works and redirects you to a success page.


**Other information**:
- The version has been updated to v1.2.0.

kleberbaum commented on src/components/pages/HomePage/index.jsx:
```diff
@@ -19,172 +19,11 @@ import {
-          </MDBContainer>
-        </div>
-      </>
+      null
```
NOTE: It would be appreciated to comment why a render function returns null.

Aichnerc replied:
Well, because nothing should be displayed. Comment will be added.

kleberbaum commented on public/fav/browserconfig.xml:
```diff
@@ -0,0 +1,2 @@
+<?xml version="1.0" encoding="utf-8"?>
+<browserconfig><msapplication><tile><square70x70logo src="/ms-icon-70x70.png"/><square150x150logo src="/ms-icon-150x150.png"/><square310x310logo src="/ms-icon-310x310.png"/><TileColor>#ffffff</TileColor></tile></msapplication></browserconfig>
```
Note: It would be much better with some line breaks.

Aichnerc replied:
Sure. Will be improved.

Aichnerc committed:
Add SmoothScrolling

Some smooth scroll links have been added.

Aichnerc committed:
Add contact functionality

It is now possible to use the contact form. Although 
updates for anon user and error messages are being 
required.

Aichnerc committed:
Improve code quality

Some unused imports and variables have been 
removed.

Aichnerc committed:
Improve footer logo

The logo is no longer that big on mobile devices.

Aichnerc committed:
Improve mobile view

Some changes have been made to improve the 
mobile experience.

Aichnerc committed:
Improve trusted

The logos are now being created dynamically.

Aichnerc committed:
Add key param to Trusted section

A missing key parameter has been added.

Aichnerc committed:
Improve mobile hero

The hero section has been improved to improve 
the mobile experience.

Aichnerc committed:
Improve ContactForm

The contact form has been improved by adding a 
thank you page.

Aichnerc committed:
Improve meta tag color

The color has been adjusted.

Aichnerc committed:
Improve SEO

The robots.txt and sitemap have been added.

Aichnerc committed:
Add React-Helmet

This allows us to set meta tags on each page.

Aichnerc committed:
Improve mobile design

The mobile view has been improved.

Aichnerc committed:
Move sitemap.xml

The sitemap has been moved to a sub folder to test 
if it can be detected.

Aichnerc committed:
Move sitemap.xml back

The subfolder was not needed for it to be found.

Aichnerc committed:
Update version

The version has been updated (minor).

Aichnerc committed:
Improve code quality

The indentation may be bugged.

Aichnerc committed:
Improve code quality

The indentation bug is really annoying,.

Aichnerc committed:
Fix indentation

The indentation bug has been fixed.

pinterid commented on src/components/organisms/sections/CallToAction/index.jsx:
```diff
@@ -2,15 +2,23 @@
 import React from "react";
 // React Router DOM bindings
-import { Link } from "react-router-dom";
+//import { Link } from "react-router-dom";
```
Is this still required?

Aichnerc replied:
Yes. Once this is only very temporary.

Aichnerc committed:
Fix indentation

The indentation has been changed on the index.js file.

pinterid commented on src/components/organisms/sections/ContactForm/index.jsx:
```diff
@@ -21,6 +20,12 @@ import {
+//> Redux
+// Connect
+import { connect } from "react-redux";
+// Actions
```
Missing new line between internal and external imports.

Aichnerc replied:
These are both Redux based imports under the //> Redux header.

pinterid commented on src/components/organisms/sections/ContactForm/index.jsx:
```diff
@@ -47,15 +52,30 @@ class ContactForm extends React.Component {
+
   render() {
-    const { darkMode } = this.props;
+    console.log(this.props);
```
Is this still required?

pinterid commented on src/components/organisms/sections/Services/index.jsx:
```diff
@@ -1,12 +1,12 @@
-// Router Link
-import { Link } from "react-router-dom";
+// Router
+//import { Link } from "react-router-dom";
```
Is this still required?

Aichnerc replied:
Yes. Once this is only very temporary.

Aichnerc commented on src/store/actions/contactActions.js:
```diff
@@ -0,0 +1,28 @@
+  return (dispatch, getState, { getFirebase, getFirestore }) => {
+    const firestore = getFirestore();
+
+    // Create contact post
```
Should be "Create new contact".

LucaAllmaier commented on src/components/organisms/sections/ContactForm/index.jsx:
```diff
@@ -33,211 +33,211 @@ import "./contactForm.scss";
+                <MDBCardBody className="py-5">
+                  <h3 className="font-weight-bold">Kontakt aufnehmen</h3>
+                  <p className="text-muted">
+                    Du hast eine Projektidee und bist Dir nicht sicher um deren
```
...bezüglich deren Umsetzung...

Aichnerc commented on src/store/actions/contactActions.js:
```diff
@@ -0,0 +1,28 @@
+export const createContact = (newContact) => {
+  return (dispatch, getState, { getFirebase, getFirestore }) => {
```
getFirebase is not required in this function.

LucaAllmaier commented on src/components/organisms/sections/ContactForm/index.jsx:
```diff
@@ -33,211 +33,211 @@ import "./contactForm.scss";
+                        type="text"
+                        className="form-control"
+                        placeholder="Deine voller Name"
+                        aria-label="Deine voller Name"
```
Deine Name?

Aichnerc replied:
Sounds better, yes.

Aichnerc reviewed (commented):
Still a few minor issues.

pinterid commented on src/index.js:
```diff
@@ -64,33 +64,35 @@ const userCollection = "partners";
-      attachAuthIsReady: onlyLoadWhenReady,
-    })
-  )
+	rootReducer,
```
Wrong tab size?

Aichnerc replied:
**Calling it tab size is missleading.**
It should not contain tabs, but spaces.
This issue has been fixed, by converting all indenty to spaces. My laptop was still using tabs instead of spaces.

sweggerat reviewed (approved):
looks good to me

pinterid commented on src/store/actions/contactActions.js:
```diff
@@ -0,0 +1,28 @@
+      })
+      .then(() => {
+        dispatch({ type: "CONTACT_SUCCESS" });
+        return;
```
Should this return undefined?

pinterid commented on src/store/reducers/rootReducer.js:
```diff
@@ -13,6 +15,7 @@ import { firebaseReducer } from "react-redux-firebase";
   auth: authReducer,
+  contact: contactReducer,
   firestore: firestoreReducer,
   firebase: firebaseReducer, // Authentication
```
Comment should not be in the same line.

Aichnerc replied:
This is okay, once it explains the content of an Object directly.

pinterid reviewed (changes requested):
Should be ok to merge after a few changes.

LucaAllmaier reviewed (approved):
Looks good now.

Aichnerc committed:
Improve code quality

The code quality has been improved due to comments by @pinterid and @LucaAllmaier.

pinterid reviewed (approved):
Looks good to merge.

schettn reviewed (approved):
LGTM!

Aichnerc committed:
Fix indentation

The indentation of a file has been fixed.

kleberbaum commented on src/components/molecules/Navbar/index.jsx:
```diff
@@ -78,7 +65,8 @@ class Navbar extends React.Component {
           color={this.props.darkMode ? "agency-dark" : "white"}
-          {...this.getMode()}
+          light={!this.props.darkMode}
+          dark={this.props.darkMode}
```
xD lightMode is !darkMode.

Aichnerc replied:
This is correct, yes. There are only two modes. Light and dark. If the mode is not dark it should be light, right?

kleberbaum reviewed (approved):
Should be ok to merge.
